src/App.tsx

```diff
@@ -15,48 +15,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.NONE:
+      break;
+
+    case SortType.ALPHABET:
+      visibleGoods.sort(
+        (good1, good2) => good1.localeCompare(good2),
+      );
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort(
+        (good1, good2) => (good1.length - good2.length),
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, sortBy] = useState(SortType.NONE);
+  const [isReversed, reverseArray] = useState(false);
+  const [isReset, resetVisible] = useState(false);
```

Comment:
no need to create the separate `isReset` state variable
you can calculate it from `isReversed` and `sortType`

Reply:
fixed